### ISSUE TITLE
Datatable: inside datatable command button not work.

### ISSUE THREAD

**Issue by lucassolerne** [primefaces/primefaces#10445]
### Describe the bug

Component: When used datatable inside datatable command button not work.

Example:

```xml
   <p:dataTable value="#{lancamentoReceberController.selected.parcelaReceberList}"
                                     style="font-size: 10px;min-width: 100%" 
                                     rowStyleClass="#{empty rowIx or rowIx mod 2 ne 0 ? 'even-row' : 'odd-row'}" 
                                     rowIndexVar="rowIx" 
                                     editable="true"
                                     rowKey="#{parcelasDetalhe.id}"
                                     id="parcelasDetalhe"
                                     var="parcela">

                           

                            <p:column headerText="Doc" style="text-align: center;text-transform: uppercase">
                                <p:cellEditor>  
                                    <f:facet name="output">  
                                        <h:outputText value="#{parcela.numeroDocumentoOrigem}-#{parcela.parcela}"/>
                                    </f:facet>  
                                    <f:facet name="input">  
                                        <p:inputText onclick="this.select()"
                                                     readonly="#{!funcaoComponenteController.habilitado('LancamentoReceberPermiteAlteracaoDocDaParcela')}"
                                                     value="#{parcela.numeroDocumentoOrigem}"/>
                                    </f:facet>
                                </p:cellEditor>
                            </p:column>


                            <p:column>
                                <p:commandButton title="Receber" ajax="true" 
                                                 process="@this"
                                                 action="#{lancamentoReceberController.setParcelaReceberSelecionadaBaixa(parcela)}"
                                                 rendered="#{parcela.statusParcela == '1'}"
                                                 update=":form:parcelas_baixar"
                                                 oncomplete="PF('baixarParcela').show()"
                                                 icon="ui-icon-gear">    
                                </p:commandButton> 
                                <p:commandButton title="Cancelar" ajax="true" 
                                                 process="@this"
                                                 onclick="PF('statusGeral').show()"
                                                 rendered="#{funcaoComponenteController.habilitado('LancamentoReceberEditBotaoCancelarParcela') and parcela.statusParcela == '1'}"
                                                 action="#{lancamentoReceberController.setParcelaReceberSelecionadaCancelar(parcela)}"
                                                 update=":form:parcelas_cancelar"
                                                 oncomplete="PF('cancelarParcela').show();PF('statusGeral').hide()"
                                                 icon="ui-icon-cancel">    
                                </p:commandButton> 
                                <p:commandButton id="downloadLink" title="Gerar Boleto para esta parcela" ajax="false" 
                                                 onclick="PrimeFaces.monitorDownload(start, stop);
                                                         this.form.target = '_self'"
                                                 rendered="#{parcela.statusParcela == '1'
                                                             and parcela.formaPagamento.tipo == 4
                                                             and parcela.convenioBancario != null
                                                             and parcela.convenioBancario.getConvenioBancarioConfiguracao(parcela.lancamentoReceber.empresa).permiteImpressaoBoleto == 'S'}"
                                                 icon="fa fa-barcode">  
                                    <p:fileDownload value="#{lancamentoReceberController.gerarBoleto(parcela)}" />  
                                </p:commandButton>            
                            </p:column>


                            <p:rowExpansion>
                                <p:tabView id="tabParcelaDetalhe" dynamic="true" cache="true">
                                    <p:tab id="abaParcelaBaixas" title="Baixas/Adiantamentos">
                                        <h:panelGrid id="gridBaixas" columns="1" styleClass="semBorda">
                                            <h:outputLabel style="font-weight: bold" value="Baixas/Adiantamentos"/>
                                            <p:dataTable 
                                                lazy="true"
                                                emptyMessage="Nenhuma baixa ou adiantamento efetuado"
                                                value="#{parcela.parcelaRecebimentoList}"
                                                style="min-width: 100%; max-width: 100%;"
                                                var="itemRecebimento"
                                                paginator="false">  

                                                <p:column width="50" headerText="Tipo">
                                                    <h:outputText value="#{itemRecebimento.descricaoTipoRecebimento}"/>
                                                </p:column>

                                                

                                                <p:column width="50">
                                                    
<!--        Problem started with change for prime 11 not work. When we used primefaces 8  worked normally.
                                                   <p:commandButton 
                                                        rendered="#{itemRecebimento.cancelado == 'N'}"
                                                        ajax="false"
                                                        process="@this"
                                                        onclick="this.form.target = '_blank'"
                                                        action="#{lancamentoReceberController.imprimirReciboRecebimento(itemRecebimento)}"
                                                        icon="ui-icon-print"
                                                        title="Imprimir Recibo"/>
                                                    <meta:commandButtonCancelamento rendered="#{funcaoComponenteController.habilitado('LancamentoReceberListBotaoCancelarBaixa') 
                                                                                                and itemRecebimento.cancelado == 'N' and itemRecebimento.parcelaReceber.statusParcela != '2'}"
                                                                                    update=":messagePanel @this :form:tabParcelas:abaParcelas"
                                                                                    acao="#{lancamentoReceberController.cancelarBaixa(itemRecebimento)}"
                                                                                    motivoCancelamento="#{itemRecebimento.observacaoCancelamento}"                                                                 
                                                                                    messagem="Deseja estornar a Baixa? (Caso a parcela tenha sido baixada em lote, todas as baixas serão reabertas e suas devidas movimentações estornadas)">
                                                    </meta:commandButtonCancelamento>-->

                                                </p:column>

                                            </p:dataTable>
                                              </p:tab>
                    
                                            </p:tabView>

                                             </p:rowExpansion>
```

### Reproducer

[primefaces-test-master.zip](https://github.com/primefaces/primefaces/files/12359068/primefaces-test-master.zip)

### Expected behavior

The commandButton should work, as the one outside the rowExpansion

### PrimeFaces edition

Community

### PrimeFaces version

11.0.0

### Theme

Spark

### JSF implementation

Mojarra

### JSF version

2.3

### Java version

17

**Comment by tandraschko** [comment]
1) please try with 13.0, 11.0 is not supported by the ocmmunity
2) provide a reproducer (forked primefaces-test), otherwise we cant replicate

**Comment by fcorneli** [comment]
It is actually an interesting problem on how to correctly provide a "custom variable" as part of a JSF component. Consider the following example page:
https://github.com/e-Contract/enterprise-jsf/blob/master/examples/helloworld-ear/war/src/main/webapp/custom-variable.xhtml
Here I want the `exampleVar` variable to be available towards the children of the `customVariable` JSF component during _ALL_ JSF phases. One also has to take care of nesting. Best thing I could come up with is the following:
https://github.com/e-Contract/enterprise-jsf/blob/master/examples/helloworld-ear/taglib/src/main/java/be/e_contract/jsf/taglib/variable/CustomVariableComponent.java
To ensure that your custom variable is available during all JSF phases, the easiest thing to do is to highjack `pushComponentToEL` and `popComponentFromEL` I guess. Right?
For `UIData` based components you might need to play a bit with the actual value expression.

**Comment by lio016** [comment]
I have a same problem. But not need to be datatable in rowExpension -> CommandButton (action and actionListener) does not work in rowExpension. (onClick() is ok). If there is CommandButton in table cell, it works properly. The problem presents in more versions (10,11,12,13)
```
<p:dataTable value="#{dataList}" var="row" ...>
  <p:column>
    <h:form>
      <p:commandButton value="button"> <-- it is working
        <p:fileDownload value="#{controller.getDocument(row)}"/>
      </p:commandButton>
    </h:form>
  </p:column>
  <p:rowExpansion>
    <h:form>
      <p:commandButton value="button"> <-- does not work
        <p:fileDownload value="#{controller.getDocument(row)}"/> 
      </p:commandButton>
    </h:form>
  </p:rowExpansion>
</p:dataTable>
```

**Comment by jepsar** [comment]
@lio016 can you please provide a primefaces-test reproducer?

**Comment by lio016** [comment]
@jepsar I have attached a zip file with the reproducable code.
[primefaces-test-master.zip](https://github.com/primefaces/primefaces/files/12359068/primefaces-test-master.zip)

**Comment by melloware** [comment]
Similar to this RowExpansion issue: https://github.com/primefaces/primefaces/issues/10308

**Comment by lio016** [comment]
If I use one form around a datatable in my code the buttons work fine.
[primefaces-test-master2.zip](https://github.com/primefaces/primefaces/files/12366426/primefaces-test-master2.zip)

**Comment by jepsar** [comment]
> If I use one form around a datatable in my code the buttons work fine. [primefaces-test-master2.zip](https://github.com/primefaces/primefaces/files/12366426/primefaces-test-master2.zip)

I was doubting if that was indeed valid. I'm not sure if your (@lio016) problem is / was the same as the OP's.

**Comment by melloware** [comment]
@lucassolerne closing for now until a reproducer is provided.

**Comment by ltlBeBoy** [comment]
I recently encountered a similar issue with `p:rowExpansion`. I attempted to create a reproducer for it, but everything functioned correctly in that context. It was then that I noticed the reproducer was already running on PF 13.0.1, while my project was on PF 13.0.0. Updating to the latest version resolved the issue. I believe it was addressed by #10328.

Hint: I'm use Payara with Mojarra.

**Comment by ltlBeBoy** [comment]
Here is the reproducer.

1. Expand a row.
2. Select "Customer"
3. Enter the a value
4. Click "reset"
5. Select "Customer" -> fails with PF13.0.0, works with PF13.0.1!

[primefaces-test-master.zip](https://github.com/primefaces/primefaces/files/12621201/primefaces-test-master.zip)

**Comment by melloware** [comment]
Excellent!  Glad its fixed in 13.0.1
